Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -228,63 +243,117 @@ open Chain
 
 variable [OmegaCompletePartialOrder β]
 variable [OmegaCompletePartialOrder γ]
+variable {f : α → β} {g : β → γ}
+
+/-- ωScottContinuous - Scott Continuous over Chains-/
+def ωScottContinuous (f : α → β) := ScottContinuousOn (Set.range fun c : Chain α => Set.range c) f
+
+lemma _root_.ScottContinuous.ωScottContinuous (hf : ScottContinuous f) : ωScottContinuous f :=
+  hf.scottContinuousOn
+
+lemma ωScottContinuous.monotone (h : ωScottContinuous f) : Monotone f :=
+  ScottContinuousOn.monotone _ (fun a b hab => by
+    use pair a b hab; exact range_pair a b hab) h
+
+lemma ωScottContinuous.isLUB {c : Chain α} (hf : ωScottContinuous f) :
+    IsLUB (Set.range (c.map ⟨f, hf.monotone⟩)) (f (ωSup c)) := by
+  simpa [map_coe, OrderHom.coe_mk, Set.range_comp]
+    using hf (by simp) (Set.range_nonempty _) (isChain_range c).directedOn (isLUB_range_ωSup c)
+
+lemma ωScottContinuous.id : ωScottContinuous (id : α → α) := ScottContinuousOn.id
+
+lemma ωScottContinuous.map_ωSup (hf : ωScottContinuous f) (c : Chain α) :
+    f (ωSup c) = ωSup (c.map ⟨f, hf.monotone⟩) := ωSup_eq_of_isLUB hf.isLUB
+
+lemma ωScottContinuous_iff_monotone_map_ωSup :
```

Comment:
To prove (the old) continuous, it was definitionally a bit nicer than this because you don't have to provide `hf` (and then do an eta step), so let's add a lemma like
```
ωScottContinuous f ↔ ∀ c : Chain α, f (ωSup c) = ωSup (c.map f)
```
for functions like `α →o β`

Reply:
Done, thanks!